sklearn/preprocessing/data.py

```diff
@@ -460,23 +455,22 @@ def __init__(self, degree=2, interaction_only=False, include_bias=True):
         self.include_bias = include_bias
 
     @staticmethod
-    def _power_matrix(n_features, degree, interaction_only, include_bias):
-        """Compute the matrix of polynomial powers"""
+    def _combinations(n_features, degree, interaction_only, include_bias):
         comb = (combinations if interaction_only else combinations_w_r)
         start = int(not include_bias)
-        combn = chain.from_iterable(comb(range(n_features), i)
-                                    for i in range(start, degree + 1))
-        powers = np.vstack(bincount(c, minlength=n_features) for c in combn)
-        return powers
+        return chain.from_iterable(comb(range(n_features), i)
+                                   for i in range(start, degree + 1))
 
     def fit(self, X, y=None):
         """
-        Compute the polynomial feature combinations
+        Compute number of output features.
         """
         n_samples, n_features = check_array(X).shape
-        self.powers_ = self._power_matrix(n_features, self.degree,
+        combinations = self._combinations(n_features, self.degree,
                                           self.interaction_only,
                                           self.include_bias)
+        self.n_input_features_ = n_features
+        self.n_output_features_ = sum(1 for _ in combinations)
```

Comment:
I think it's worth providing `powers_` so the user has a description of each feature. Alternatively, `powers_` could be calculated dynamically as a `@property` to avoid the `fit` overhead.

You could calculate `n_output_features_` using stats.special.binom if you cared to. But I gather the current implementation is no big deal.